### PR TITLE
Plugins page and other fixes

### PR DESCRIPTION
**Pull request by uF4No**
Renames "Reference" page to "Plugins".
Improves description of configuration fields like `compilerSource`.
Typos.
Fixes typo in AA tutorial
Adds more content in tutorials index page

### CODE REVIEW

File: docs/api/hardhat/getting-started.md [hunk 1]
```diff
@@ -2,28 +2,27 @@
 
 [Hardhat](https://hardhat.org) is an Ethereum development environment, designed for easy smart contract development in Solidity. One of its most prominent features is extendability: you can easily add new plugins to your hardhat project.
 
-zkSync has three plugins for hardhat:
+zkSync has three plugins for Hardhat:
 
-- `@matterlabs/hardhat-zksync-solc` for smart contract compilation using Solidity.
-- `@matterlabs/hardhat-zksync-vyper` for smart contract compilation using Vyper.
-- `@matterlabs/hardhat-zksync-deploy` for smart contract deployment.
+- [@matterlabs/hardhat-zksync-solc](./plugins.md#matterlabs-hardhat-zksync-solc) for smart contract compilation using Solidity.
```
Comment (githubdoramon) [inline docs/api/hardhat/getting-started.md:7]: Should we actually write "for compiling smart contracts written I'm solidity"? And same for vyper

File: docs/api/hardhat/plugins.md [hunk 2]
```diff
@@ -106,19 +113,27 @@ networks: {
...
-- `compilerPath` is a field with the path to the `zkvyper` binary. If `compilerSource` is `docker`, this field is ignored. By default, the binary in `$PATH` is used.
+- `compilerSource` indicates the compiler source and can be either `docker` or `binary` (recommended). If there's not a compiler binary already installed, the plugin will automatically download it. If `docker` is used, you'd need to run Docker desktop in the background and provide both `dockerImage` and `tag` in the experimental section.
+- `compilerPath` is a field with the path to the `zkvyper` binary. By default, the binary in `$PATH` is used. If `compilerSource` is `docker`, this field is ignored.
 - `dockerImage` and `tag` make up the name of the compiler docker image. If `compilerSource` is `binary`, these fields are ignored.
-- `zksync` network option indicates whether zksolc is enabled on a certain network. `false` by default.
-
+- `libraries` if your contract uses non-inlinable libraries as dependencies, they have to be defined here. Learn more about [compiling libraries here](./compiling-libraries.md)
+- `zksync` network option indicates whether zkvyper is enabled on a certain network. `false` by default.
```
Comment (githubdoramon) [inline docs/api/hardhat/plugins.md:120]: What do this actually means? Of I set to false, or doesn't compile?

File: docs/api/hardhat/testing.md [hunk 3]
```diff
@@ -60,14 +65,15 @@ The full list of the addresses of these accounts with the corresponding private
 
 The initial version of the local node was shipped with several ERC-20 tokens deployed by default.
 
-It's no longer the case and if you need to interact with ERC-20 tokens, you should deploy them yourself.
+That's no longer the case and **if you need to interact with ERC-20 tokens, you should deploy them yourself.**
```
Comment (githubdoramon) [inline docs/api/hardhat/testing.md:68]: Should we maybe just remove the info about erc20s? Or move to a last section of troubleshoot or decpracted things? Just so we don't add an info that is only valid for a small set of people given the large Dev base will be new

**Review by githubdoramon — COMMENTED** [review]
Some minor comments... Not blockers thought, just opinions/suggestions

File: docs/api/hardhat/getting-started.md [hunk 4]
```diff
@@ -2,28 +2,27 @@
...
+- [@matterlabs/hardhat-zksync-vyper](./plugins.md#matterlabs-hardhat-zksync-vyper) for smart contract compilation using Vyper.
+- [@matterlabs/hardhat-zksync-deploy](./plugins.md#matterlabs-hardhat-zksync-deploy) for smart contract deployment.
```
Comment (krofax) [inline docs/api/hardhat/getting-started.md:9]: Can you add a hyphen(-) and add some prefix actionable words(e.g - Is used ....)?
E.g 
- [@matterlabs/hardhat-zksync-solc](./plugins.md#matterlabs-hardhat-zksync-solc) - Is used for smart contract compilation using Solidity.
- [@matterlabs/hardhat-zksync-vyper](./plugins.md#matterlabs-hardhat-zksync-vyper) - Is used for smart contract compilation using Vyper.
- [@matterlabs/hardhat-zksync-deploy](./plugins.md#matterlabs-hardhat-zksync-deploy) - Is used for smart contract deployment.

File: docs/api/hardhat/plugins.md [hunk 5]
```diff
@@ -44,30 +47,32 @@ networks: {
...
-This plugin is used to provide a convenient interface for compiling zkSync smart contracts using Vyper.
+This plugin is used to provide a convenient interface for compiling Vyper smart contracts before deploying them to zkSync 2.0.
 
 ### Npm
 
 [@matterlabs/hardhat-zksync-vyper](https://www.npmjs.com/package/@matterlabs/hardhat-zksync-vyper)
 
 This plugin is used in conjunction with [@nomiclabs/hardhat-vyper](https://www.npmjs.com/package/@nomiclabs/hardhat-vyper).
-To use it, you have to install and import both:
+To use it, you have to install and import both in the `hardhat.config.ts` file:
```
Comment (krofax) [inline docs/api/hardhat/plugins.md:71]: Can you rephrase it to :
"To use it, you have to install and import both plugins in the `hardhat.config.ts` file:"

File: docs/api/hardhat/plugins.md [hunk 6]
```diff
@@ -106,19 +113,27 @@ networks: {
 ```
 
 - `version` is a field with the version of the `zkvyper` compiler. Currently not used.
-- `compilerSource` is a field with the compiler source. It can be either `docker` or `binary`. In the former case, the `dockerImage` value with the name of the compiler docker image should also be provided. In the latter case, the compiler binary should be provided in `$PATH`.
-- `compilerPath` is a field with the path to the `zkvyper` binary. If `compilerSource` is `docker`, this field is ignored. By default, the binary in `$PATH` is used.
+- `compilerSource` indicates the compiler source and can be either `docker` or `binary` (recommended). If there's not a compiler binary already installed, the plugin will automatically download it. If `docker` is used, you'd need to run Docker desktop in the background and provide both `dockerImage` and `tag` in the experimental section.
```
Comment (krofax) [inline docs/api/hardhat/plugins.md:116]: Can you rephrase it to :
If no compiler binary was installed ..... or If there isn't a compiler binary already installed ....

File: docs/.vuepress/config.js [hunk 7]
```diff
@@ -71,13 +71,7 @@ module.exports = {
...
           path: "/dev/fundamentals",
           collapsable: false,
-          children: [
-            "/dev/fundamentals/rollups.md",
-            "/dev/fundamentals/zkSync.md",
-            "/dev/fundamentals/testnet.md",
-            "/dev/fundamentals/faq.md",
-          ],
-
+          children: ["/dev/fundamentals/rollups.md", "/dev/fundamentals/zkSync.md", "/dev/fundamentals/testnet.md", "/dev/fundamentals/faq.md"],
```
[the remark this answers is not in the document]
Comment (githubdoramon) [reply docs/.vuepress/config.js:74]: I need eslint rules so personal settings do not change PRs. I don't mind any, as long as everyone follows the same automatically

[the remark this answers is not in the document]
Comment (uF4No) [reply docs/.vuepress/config.js:74]: that's actually changed on save by the code editor 😕

File: docs/api/hardhat/testing.md (hunk 3 above)
Comment (uF4No) [reply docs/api/hardhat/testing.md:68]: I think it's important to mention ERC20s, even more if this was a feature in the past.  I've simplified the message and leave it as a warning in the rich wallets section. I think that's the most appropriate place to have it for now

File: docs/api/hardhat/getting-started.md (hunk 1 above)
Comment (uF4No) [reply docs/api/hardhat/getting-started.md]: fixed in e28f4343571352dec4b23cfb303616a2f8f497fa

File: docs/api/hardhat/plugins.md (hunk 2 above)
Comment (uF4No) [reply docs/api/hardhat/plugins.md]: fixed in e28f4343571352dec4b23cfb303616a2f8f497fa after talking with @ly0va

File: docs/api/hardhat/testing.md (hunk 3 above)
Comment (uF4No) [reply docs/api/hardhat/testing.md]: fixed in e28f4343571352dec4b23cfb303616a2f8f497fa

File: docs/api/hardhat/plugins.md (hunk 6 above)
Comment (uF4No) [reply docs/api/hardhat/plugins.md]: fixed in e28f4343571352dec4b23cfb303616a2f8f497fa

**Review by githubdoramon — APPROVED** [review]
